### ISSUE TITLE
Qt Crashes when creating a new package

### ISSUE THREAD

**Issue by mikearmstrong800** [ros-industrial/ros_qtc_plugin#79]
I just installed the plug-in and all seemed to go well.  But when I attempt to create a new ROS package Qt  crashes when I click finish.  I'm running Ubuntu 16.04.

It did work once when I put dummy items in the depends fields.  But only once.  Also, I made a sample URDF file without any problems.

This looks like an excellent product.  Any help would be appreciated.

Sincerely,

Mike Armstrong

**Comment by Levi-Armstrong** [comment]
I will test it on 14.04 and see if it crashes.

**Comment by Levi-Armstrong** [comment]
Have you updated recently?

**Comment by mikearmstrong800** [comment]
I'm a first time user.  I installed it from this tutorial -> https://github.com/ros-industrial/ros_qtc_plugin/wiki/1.-How-to-Install-(Users).

Although the video is different than my version, I was able to follow the tutorial to create a catkin workspace/package/node.

When I got to the second video, setting up Qt build, that is substantially different, and there are no predefined build arguments like in the video.  I'm not sure what's going on.

What I'm looking to do at this point is have a basic GUI for say, a listener node.

Thanks for all your hard work.

Any suggestions on how I should proceed?

Mike

**Comment by Levi-Armstrong** [comment]
I just updated the wiki today, to prevent further dependency issues. I would recommend going back there the wiki. I would also recommend going through the wiki "Setup Qt Creator for ROS". I would not bother with the 3.5 and 3.6 at this point, but maybe in the future.

I apologize about the videos. I just updated everything from 4.1 to 4.2 and things in Qt Creator have changed so I plan to update the tutorial videos soon.

>What I'm looking to do at this point is have a basic GUI for say, a listener node.

Let me know if I understand you correctly. You are wanting to create a Qt GUI with in your ROS package, along with a node in your pacakge?

**Comment by mikearmstrong800** [comment]
I already have a lot of time invested in a C++ Qt GUI and controller for a 6 DOF robot, so I’d like to stick with Qt if possible.  What I want to do immediately is enter position and orientation values into QlineEdit’s and press a QPushButton to publish them to a Moveit! based node.  From that node my Qt app will have a subscriber node to receive trajectory information that will be displayed in a QTextBrowser and be sent to the controller.

So now, I’m in the process of getting up to speed on using Qt widgets and ROS.  I’m also looking into writing an rqt plugin, but as I stated above, I already have the Qt stuff in place.

If you know of where I can find working example code for what I describe I’d appreciate it.

I looked over the wiki changes.  Thank you.

Mike

**Comment by Levi-Armstrong** [comment]
I assume by "C++ Qt GUI and controller for a 6 DOF robot" you are currently using qmake to build these items.  I would recommend porting your build system over to catkin_make (cmake) since you are wanting to use ROS components. You can find a lot of information on the web on converting qmake to cmake. From here everything should be following the ROS example on publisher and subscribers.

**Comment by Levi-Armstrong** [comment]
Here is a useful [resource](https://github.com/stonier/qt_ros/tree/indigo/qt_tutorials/common)

**Comment by Levi-Armstrong** [comment]
Also if the code is in a public GitHub repository, I could take a look at it to get a better understanding of what you are trying to do?

**Comment by mikearmstrong800** [comment]
Levi,

I finally got my original Qt C++ controller/GUI code "catkinized" and working in Qt with your plugin.

I did have it crash again when I created the package, so I just made one from the command line.  I'm not too worried about that now.

Thank you for all your help and offer to look at my code.  It's not on any public repositories.

In the end it wasn't as bad as it seemed from reading all of the accounts on line.  There are really good manuals for using Qt Widgets with CMake.  The links are below for reference.

Thanks again,
Mike

links:
https://cmake.org/cmake/help/v3.0/manual/cmake-qt.7.html

http://doc.qt.io/qt-5/cmake-manual.html
